grpc/enums.go

```diff
@@ -2,23 +2,23 @@ package grpc
 
 import "github.com/buildwithgrove/path/envoy/auth_server/proto"
 
-// These enums exist to simplify the usage of converting from data source types to
+// DEV_NOTE: These enums exist to simplify the usage of converting from data source types to
 // the types expected by the proto package.
 
 // AuthType is a string type representing the authorization type for a gateway
 // endpoint, which matches the values in the AuthType enum in the proto package.
 type AuthType string
 
 const (
-	AuthTypeNoAuth AuthType = "NO_AUTH"
-	AuthTypeAPIKey AuthType = "API_KEY_AUTH"
-	AuthTypeJWT    AuthType = "JWT_AUTH"
+	AuthTypeNoAuth AuthType = "AUTH_TYPE_UNSPECIFIED"
```

Comment:
Shouldn't this be something like `AUTH_TYPE_UNSECURED` ?

Reply:
`UNSPECIFIED` is the idiomatic way to label empty enum type fields as defined by the protobuf best practices. See here: https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum

Reply:
+1 to have the `0` field be `UNSPECIFIED`.

It's the best practice that everyone uses (and hates) at Google, but bears fruits in the long-term.

Intuition leads us to believe to use a proper default, but having unspecified as the default helps catch issues more readly.